pkg/machines/libvirtUtils.js

```diff
@@ -67,7 +67,7 @@ export function prepareNICParam(nics) {
             user: nic.type === "user",
             bridge: nic.source.bridge,
             network: nic.source.network,
-            type: nic.type === "direct" ? "direct" : null,
+            type: (nic.type === "direct" || nic.type === "ethernet") ? nic.type : null,
             source: nic.source.dev,
             mac: nic.mac,
             model: nic.model,
```

Comment:
I mean these options.

Reply:
if you check in parseDumpxmlForInterfaces function how we construct the NIC xml, the `source` key for example always exists, so source.dev can be undefined, but it will not be a problem to access it. For this reason, all of the above that will be undefined will just be ignored.
(I actually tried it)